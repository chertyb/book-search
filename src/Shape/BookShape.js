import PropTypes from 'prop-types';

const BookShape = PropTypes.shape({
  id: PropTypes.string,
  title: PropTypes.string,
  authors: PropTypes.string,
  categories: PropTypes.string,
  description: PropTypes.string,
  thumbnail: PropTypes.string,
});

export const defaultBookShape = {
  id: '',
  title: '',
  authors: '',
  categories: '',
  description: '',
  thumbnail: '',
};

export default BookShape;
