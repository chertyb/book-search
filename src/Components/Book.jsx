import React from 'react';
import BookShape, { defaultBookShape } from '../Shape/BookShape';

const Book = (props) => {
  const { data } = props;

  return (
    <div className="book">
      <img src={data.thumbnail} alt="thumbnail" />
      <p title="Title">
        {'Title: '}
        {data.title}
      </p>
      <p title="Authors">
        {'Authors: '}
        {data.authors}
      </p>
      <p title="Categories">
        {'Categories: '}
        {data.categories}
      </p>
      <p title="Description">
        {'Description: '}
        {data.description}
      </p>
    </div>
  );
};

Book.propTypes = {
  data: BookShape,
};

Book.defaultProps = {
  data: defaultBookShape,
};

export default Book;
