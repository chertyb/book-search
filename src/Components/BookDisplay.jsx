import React from 'react';
import PropTypes from 'prop-types';
import Book from './Book';
import BookShape from '../Shape/BookShape';

const BookDisplay = (props) => {
  const { books, totalBooks, loadMore } = props;

  const handleScroll = (e) => {
    const divBottom = e.target.scrollHeight - e.target.scrollTop === e.target.clientHeight;
    if (divBottom && books.length < totalBooks) {
      loadMore(false);
    }
  };

  return (
    <div id="book-display" onScroll={handleScroll}>
      {books
      && books.map((book) => <Book data={book} key={`book-${book.id}`} />)}
    </div>
  );
};

BookDisplay.propTypes = {
  books: PropTypes.arrayOf(BookShape),
  loadMore: PropTypes.func,
  totalBooks: PropTypes.number,
};

BookDisplay.defaultProps = {
  books: [],
  loadMore: () => {},
  totalBooks: 0,
};

export default BookDisplay;
