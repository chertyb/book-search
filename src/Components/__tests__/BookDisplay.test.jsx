import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from '../../enzyme';
import Book from '../Book';
import BookDisplay from '../BookDisplay';
import { defaultBookShape } from '../../Shape/BookShape';

describe('BookDisplay component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<BookDisplay />, div);
    expect(div).toMatchSnapshot();
    ReactDom.unmountComponentAtNode(div);
  });

  it('should display books', () => {
    const books = [defaultBookShape, defaultBookShape, defaultBookShape];
    const wrapper = shallow(<BookDisplay books={books} totalBooks={books.length} />);
    expect(wrapper.find(Book).length).toEqual(books.length);
  });
});
