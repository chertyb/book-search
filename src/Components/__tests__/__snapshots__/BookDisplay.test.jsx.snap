// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDisplay component renders without crashing 1`] = `
<div>
  <div
    id="book-display"
  />
</div>
`;
